src/App.tsx

```diff
@@ -1,39 +1,68 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type Props = {};
+type State = {
+  hasClock: boolean,
+  clockName: string
+};
+
+export class App extends React.Component<Props, State> {
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  timerId = 0;
+
+  componentDidMount() {
+    document.addEventListener('contextmenu', this.handleRightClick);
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+    this.timerId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+  }
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  componentDidUpdate() {
+    document.addEventListener('click', this.handleLeftClick);
+  }
```

Comment:
why not add listener on `DidMount`?

Reply:
IDK hehe I notice now that it could be done all in the same method! all done!